data_model/src/events/time.rs

```diff
@@ -353,13 +387,36 @@ mod tests {
             // ----[-*-*-*-*-*-*-)-*-*-*
             //   p, i1           i2
 
-            let schedule =
-                Schedule::new(Duration::from_secs(TIMESTAMP), Some(Duration::from_secs(1)));
+            let schedule = Schedule::starting_at(Duration::from_secs(TIMESTAMP))
+                .with_period(Duration::from_secs(1));
             let interval = Interval::new(Duration::from_secs(TIMESTAMP), Duration::from_secs(7));
             assert_eq!(
                 EventFilter::count_matches_in_interval(&schedule, &interval),
                 7
             );
         }
+
+        #[test]
+        fn bench_schedule_from_zero_with_little_period() {
+            //                       *         *
+            // --|-*-*-*- ... -*-*-*-[-*-...-*-)-*-*-*-
+            //   p     52 years     i1   1sec  i2
+
+            let start_time = current_time();
+
+            let schedule =
+                Schedule::starting_at(Duration::ZERO).with_period(Duration::from_millis(1));
+            let interval = Interval::new(Duration::from_secs(TIMESTAMP), Duration::from_secs(1));
+
+            assert_eq!(
+                EventFilter::count_matches_in_interval(&schedule, &interval),
+                1000
+            );
+
+            let finish_time = current_time();
+            let result_time = finish_time.saturating_sub(start_time);
+
+            assert!(result_time < Duration::from_millis(1));
+        }
```

Comment:
So do you mean, that benchmark shouldn't fail if algorithm is incorrect but takes acceptable time to work?
Makes sense... Since 1'st assertion is already tested in other tests I'll just remove it

Reply:
I mean test results should not depend on machine specs.
And benchmarks should be able to detect errors as well as tests.
I think you can create something similar to those under `benches/` or remove this as you said